RELEASING.md

```diff
@@ -4,24 +4,6 @@
 
 Create a `Version Release` issue to track the release process.
 
-## Semantic Convention Generation
```

Comment:
This shouldn't be completely removed. We still need the option to generate the packages locally and that process should be documented. Additionally, the automated way of generating the package should be added.